corehq/form_processor/tests/utils.py

```diff
@@ -172,8 +172,36 @@ def _delete_all(db, viewname, **view_kwargs):
 )
 
 
+def partitioned(cls):
+    """
+    Marks a test to be run with the partitioned database settings in
+    addition to the non-partitioned database settings.
+    """
+    return attr(sql_backend=True)(cls)
+
+
+def only_run_with_non_partitioned_database(cls):
+    """
+    Only runs the test with the non-partitioned database settings.
+    """
+    if settings.USE_PARTITIONED_DATABASE:
+        return nottest(cls)
+
+    return cls
+
+
+def only_run_with_partitioned_database(cls):
```

Comment:
What's the need for these? Is it basically the same as https://github.com/dimagi/commcare-hq/blob/d50fdf17133c68881b3600d025641a7402d524ae/corehq/form_processor/tests/test_sharding.py#L20-L20